client/interface.go

```diff
@@ -18,6 +19,22 @@ type APIClient interface {
 	CheckpointCreate(ctx context.Context, container string, options types.CheckpointCreateOptions) error
 	CheckpointDelete(ctx context.Context, container string, checkpointID string) error
 	CheckpointList(ctx context.Context, container string) ([]types.Checkpoint, error)
+	SwarmInit(ctx context.Context, req swarm.InitRequest) (string, error)
```

Comment:
The `APIClient` interface is getting bigger and bigger. We might want to split it into several ones (like `VolumeAPIClient`, `NetworkAPIClient`, `SwarmAPIClient`, …) and have `APIClient` embedded them… 

It's a nit and not in the scope of the PR… just thinking outloud.

*This would help project using `engine-api` to have less work to do to unit-test some part :angel:*

@calavera @vieux @icecrime @stevvooe WDYT ?